java/com/tigervnc/vncviewer/FileUtils.java

```diff
@@ -56,21 +58,47 @@ public static final String getHomeDir() {
       vlog.error("Cannot access os.name system property:"+e.getMessage());
     }
 
-    String separator = null;
-    try {
-      separator = Character.toString(java.io.File.separatorChar);
-    } catch(java.security.AccessControlException e) {
-      vlog.error("Cannot access file.separator system property:"+e.getMessage());
+    return homeDir + getFileSeparator();
+  }
+
+  public static String getVncDir(String xdgEnv, String xdgDefault) {
+    File legacyDir = new File(getHomeDir() + ".vnc" + getFileSeparator());
+    String os = System.getProperty("os.name");
+
+    if (os.startsWith("Windows")) {
+      File newDir = new File(System.getenv("APPDATA") + getFileSeparator() + "vnc" + getFileSeparator());
```

Comment:
We should probably use "tigervnc" instead of "vnc" here, both for consistency with what the native viewer is doing as well as so we don't potentially collide with another VNC client

Reply:
The native viewer still uses `%APPDATA%/vnc` on Windows. I had assumed all this time that that didn't really require changing, although if needed I can revisit this PR to also migrate to `tigervnc` on Windows.

Reply:
Hmm.  That might be best because after thinking about the collision scenario a bit more, if another product uses the legacy dirs, this change would cause (a) their settings to get migrated on Windows or (b) on other platforms the path to the legacy dir to be returned even if the new path exists.

At a minimum maybe it should be altered to check if the new path exists first, and if so return that before attempting a migration or issuing a warning?  I think there are scenarios where this might occur even with TigerVNC, like if someone tried using an older version for debugging a regression, etc.

Reply:
Indeed, it seems like it wasn't the best decision to prioritise the legacy path altogether if it exists. I'll instead make it fall back to `~/.vnc` if and only if the new path doesn't already exist and the legacy path does.